xbmc/peripherals/addons/AddonButtonMapping.cpp

```diff
@@ -28,11 +28,12 @@ using namespace JOYSTICK;
 using namespace PERIPHERALS;
 
 #ifndef SAFE_DELETE
-  #define SATE_DELETE(x)  do { delete (x); (x) = NULL; } while (0)
+  #define SATE_DELETE(x)  do { delete (x); (x) = nullptr; } while (0)
```

Comment:
SA**T**E_DELETE? I believe you mean "UNSAFE_DELETE";) Please use smart pointers. The memory management in this class is broken. It will leak and double free.